src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -22,9 +23,13 @@ public class DataAnnotationsMetadataProvider :
         IValidationMetadataProvider
     {
         private readonly IStringLocalizerFactory _stringLocalizerFactory;
+        private readonly IOptions<MvcDataAnnotationsLocalizationOptions> _options;
```

Comment:
Save the `MvcDataAnnotationsLocalizationOptions`. Only use `IOptions<TOptions>` instances for the constructor injection. Realize that's not the pattern used specifically where `DataAnnotationLocalizerProvider` is read today but should still do the simple thing here.